client/app/components/SearchBar/SearchBar.js

```diff
@@ -0,0 +1,76 @@
+import { Component, PropTypes } from 'react';
+import { form, input, button, span } from 'r-dom';
+
+import css from './SearchBar.css';
+import icon from './images/search-icon.svg';
+
+const SEARCH_MODE_KEYWORD = 'keyword';
+const SEARCH_MODE_LOCATION = 'location';
+const SEARCH_MODE_KEYWORD_AND_LOCATION = 'keyword-and-location';
+const SEARCH_MODES = [
+  SEARCH_MODE_KEYWORD,
+  SEARCH_MODE_LOCATION,
+  SEARCH_MODE_KEYWORD_AND_LOCATION,
+];
+
+class SearchBar extends Component {
+  constructor(props) {
+    super(props);
+
+    // The values of the inputs are synced to the component state
+    // until the form is submitted.
+    this.state = {
+      keyword: '',
+      location: '',
+    };
+  }
```

Comment:
I guess, this is good solution now, but I think that, in the future, we need to add location and keyword to redux store. (I.e. the future map component probably wants to listen location info too.)